### DISCUSSION TITLE
How to Implement Event Filter in event subscription

### DISCUSSION THREAD

**Discussion by arunkumar790** [FreeOpcUa/opcua-asyncio#1498] [Q&A] [answered]
I am developing client and trying to subscribe event with event filter option
I don't know how to implement it, My expectation is using the event filter receive only specific attributes in the event notification. Like i want to receive only "EventId", "SourceNode","Severity" but not all attributes
Could anyone help, any example would be great

obj = await client.nodes.root.get_child(["0:Objects", "2:MyObject"])
myevent = await client.nodes.root.get_child(["0:Types", "0:EventTypes", "0:BaseEventType", "2:MyEvent"])
evfilter = ???

handle = await self.subscription.subscribe_events(obj, myevent, evfilter)

Thanks

**Comment by kevinPolo** [comment, marked as answer]
You can import the `get_filter_from_event_type` function from events.py:

```Python
from asyncua.common.events import get_filter_from_event_type
...
myevent = await client.nodes.root.get_child(["0:Types", "0:EventTypes", "0:BaseEventType", "2:MyEvent"])
event_filter = await get_filter_from_event_type([myevent])
```

`event_filter` is an `asyncua.ua.EventFilter` object with 'Select' clauses and 'Where' clauses: 

[https://reference.opcfoundation.org/v105/Core/docs/Part4/7.22.3](vscode-file://vscode-app/c:/Users/kevin/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html)

You can then delete some of the `Select` clauses and subscribe with this reduced filter.

```
perform deletion by comparing the browse path or something else..
```

```Python
handle = await self.subscription.subscribe_events(obj, myevent, event_filter)
```

Or directly create the event_filter object with the `Select` clauses of `SimpleAttributeOperand` type (list).

**Reply by arunkumar790** [reply]
@kevinPolo Thanks a lot, these two approaches will be helpful
